File: examples/AdditionClientComponent_151/src/urn_jaus_example_addition_client_1_0/InternalEvents/InitToReadyEventDef.cpp
#include "urn_jaus_example_addition_client_1_0/InternalEvents/InitToReadyEventDef.h"

namespace urn_jaus_example_addition_client_1_0
{

/**
 * Returns the size of memory the used data members of the class occupies.
 * This is not necessarily the same size of memory the class actually occupies.
 * Eg. A union of an int and a double may occupy 8 bytes. However, if the data
 *     stored is an int, this function will return a size of 4 bytes.
 * 
 * @return
 */
const unsigned int InitToReadyEventDef::getSize()
{
	unsigned int size = 0;
	
	
	return size;
}

void InitToReadyEventDef::encode(unsigned char *bytes)
{
	
	if (bytes == NULL)
	{
		return;
	}
	
	int pos = 0;
}

void InitToReadyEventDef::decode(const unsigned char *bytes)
{
	
	if (bytes == NULL)
	{
		return;
	}
	
	int pos = 0;
}

InitToReadyEventDef::InitToReadyEventDef()
{
	m_Name = "InitToReadyEventDef";
}

InitToReadyEventDef::~InitToReadyEventDef()
{
}


}

File: examples/CSharp/AdditionClientComponent_151/src/urn_jaus_example_addition_client_1_0/InternalEvents/InitToReadyEventDef.cs
using JTS;
using System;
using System.Collections;

namespace urn_jaus_example_addition_client_1_0
{

public class InitToReadyEventDef : InternalEvent
{
	/**
	 * Returns the number of bytes the used data members of the class occupies
	 * in the buffer. This is not the number of bytes the data type occupies in 
	 * Java, but the bytes expected on the wire.
	 * 
	 * @return
	 */
	public override int getSize()
	{
		int size = 0;
		
		
		return size;
	}
	
	public override void encode(byte[] bytes, int pos)
	{
		
		if (bytes == null)
		{
			return;
		}
		
		
	}
	
	public override void decode(byte[] bytes, int pos)
	{
		
		if (bytes == null)
		{
			return;
		}
		
		if(pos < bytes.Length)
		{
			for(int i = pos; i<bytes.Length; i++)
			{
				bytes[i] = 0;
			}
		}
	}
	
	public InitToReadyEventDef()
	{
		m_Name = "InitToReadyEventDef";
	}
	
}

}